src/K1DModel/tendency.jl

```diff
@@ -393,3 +412,179 @@ end
 
     return dY
 end
+
+@inline function advection_tendency!(::CO.PrecipitationP3, dY, Y, aux, t)
+    FT = eltype(Y.ρq_tot)
+    # TODO - flux magnitude at top should agree
+    # with ρ * q instead of being only based on q
+
+    # P3 advection (introducing particles through boundary):
+    # TODO - change run_KiD_simulation so that temp. variables
+    # (_q_flux, _N_flux, etc) are not hard coded in the advection tendency
+
+    # update aux with state variables
+    (; ρ) = aux.thermo_variables
+    (; ρq_tot, ρq_liq, ρq_rai, ρq_ice, ρq_rim, ρq_liqonice, N_ice, N_rai, N_aer, N_liq, B_rim) = aux.microph_variables
+    @. ρq_ice = Y.ρq_ice
+    @. ρq_liq = Y.ρq_liq
+    @. ρq_rim = Y.ρq_rim
+    @. ρq_liqonice = Y.ρq_liqonice
+    @. N_ice = Y.N_ice
+    @. B_rim = Y.B_rim
+
+    # choose flux characteristics
+    _q_flux = FT(0.65e-4)
+    _N_flux = FT(40000)
+    _F_rim = FT(0.2)
+    _F_liq = FT(0.2)
+    _ρ_r_init = FT(900)
```

Comment:
These should not be hardcoded

Reply:
Right—thanks. Would it be acceptable to add them as elements of the PrecipitationP3 struct that can be initialized in run_KiD_simulation? If so, would the best place be to add them as command line arguments in parse_commandline? Otherwise, I could add them in aux as command line arguments as well. Maybe a command line argument that is used in aux could be the cleanest way to do it?

Reply:
I think they would fit better in the case setup. Since they only come about because the case specifies the top fluxes. In an actual simulation we would make sure that our domain is high enough that we don't have to worry about anything else falling in

Reply:
Ah, I see your reasoning: just to make sure, do you mean this [function](https://github.com/CliMA/KinematicDriver.jl/pull/185/files#diff-e11ad7cffaff946aac755470d2db05dbc518553d52a499dd9134587cb701e70bR285)? Maybe I can add the flux characteristics hardcoded into the initial condition, and then pass them to aux this way so that they can be used in the advection tendency? If I do this, then I can make it so that the particles are either initialized with ice_bump in the initial condition or introduced through the boundary. Or I can add them as command line arguments that are used by the initial condition. Do either of you have a preference as to which might be better?